hewr/R/score_hubverse.R

```diff
@@ -0,0 +1,88 @@
+#' defaults are `"location"`, `"reference_date"`, and
+#' `"value"`, respectively, to reflecting the format used
+#' in pyrenew-hew pipeline scripts such as
+#' `pipelines/create_observed_data_table.py`.
+#'
+#' @param forecast forecasts, as a hubverse-format
+#' [`tibble`][tibble::tibble()] produced by
+#' [to_epiweekly_quantile_table()], with columns
+#' `target_end_date`, `value`, and `horizon`
+#' @param observed observations, as a [`tibble`][tibble::tibble()].
+#' @param horizons Horizons to score. Default `c(0, 1)`
+#' @param transform transformation passed as the
+#' `fun` argument to [scoringutils::transform_forecasts()].
+#' Default [scoringutils::log_shift()].
+#' @param offset Offset for the transform passed to
+#' [scoringutils::transform_forecasts()]
+#' transforming forecasts for scoring. Default 1.
+#' @param observed_value_column Name of the column containing
+#' observed values in the `observed` table, as a string.
+#' Default `"value"`
+#' @param observed_location_column Name of the column containing
+#' location values in the `observed` table, as a string.
+#' Default `"location"`
+#' @param observed_date_column Name of the column containing
+#' date values in the `observed` table, as a string.
+#' Default `"reference_date"`
+#' @param ... Other keyword arguments passed to
+#' [scoringutils::transform_forecasts()].
+#' @return A table of scores, as the output of
+#' [scoringutils::score()], filtered to include only the
+#' transformed_scale.
+#' @export
+score_hubverse <- function(forecast,
+                           observed,
+                           horizons = c(0, 1),
+                           transform = scoringutils::log_shift,
+                           offset = 1,
+                           observed_value_column = "value",
+                           observed_location_column = "location",
+                           observed_date_column = "reference_date",
+                           ...) {
+  obs <- observed |>
+    dplyr::select(
+      location = .data[[observed_location_column]],
+      target_end_date = .data[[observed_date_column]],
+      observed = .data[[observed_value_column]]
+    )
+
+  to_score <- forecast |>
+    dplyr::filter(.data$horizon %in% !!horizons) |>
+    dplyr::inner_join(obs,
+      by = c(
+        "location",
+        "target_end_date"
+      )
+    ) |>
+    scoringutils::as_forecast_quantile(
+      predicted = "value",
+      observed = "observed",
+      quantile_level = "output_type_id"
+    ) |>
+    scoringutils::transform_forecasts(
+      fun = transform,
+      offset = offset,
+      ...
+    )
+
+  interval_coverage_95 <- purrr::partial(
+    scoringutils::interval_coverage,
+    interval_range = 95
+  )
+
+  scored <- to_score |>
+    scoringutils::score(
+      metrics = c(
+        scoringutils::get_metrics(to_score),
+        list(interval_coverage_95 = interval_coverage_95)
+      )
+    ) |>
+    dplyr::filter(.data$scale == "log")
```

Comment:
Will the log scale exist if you don't do some sort of log transform?

Reply:
Good point. Yikes.

Reply:
Fixed by bf08214